lib/pbench/test/functional/agent/cli/commands/tools/test_clear_tools.py

```diff
@@ -165,48 +199,147 @@ def test_clear_tools_test69(monkeypatch, agent_config, pbench_run, pbench_cfg):
+        b"default\n" in err
+    )
     assert exitcode == 0
+
+
+@pytest.mark.parametrize(
+    "groups",
+    ["", "default", "default,group1", "default,group1,group2", "group1,group2,group3"],
+)
+@pytest.mark.parametrize(
+    "remotes",
+    ["", "remote1", "remote1,remote2", "remote1,remote2,remote3", "remote2,remote4"],
+)
+@pytest.mark.parametrize(
+    "tools",
+    ["", "pidstat", "pidstat,mpstat", "pidstat,mpstat,iostat", "mpstat,vmstat"],
+)
+def test_clear_tools_test85(pbench_run, tools_configuration, groups, remotes, tools):
+    """ Attempt to clear tools with various combinations of groups, remotes
+    and tools against fixed tools configuration"""
+    tools_existence = tools_configuration
+
+    command = ["pbench-clear-tools"]
+    if groups:
+        command.append(f"--groups={groups}")
+    if remotes:
+        command.append(
+            f"--remotes={','.join(map('{}.example.com'.format, remotes.split(',')))}"
+        )
+    if tools:
+        command.append(f"--name={tools}")
+
+    out, err, exitcode = pytest.helpers.capture(command)
+
+    assert b"" == out
+    for tool in tools_existence:
+        tool_str = str(tool).replace(str(pbench_run), "")
+        remote_check = True
+        tool_check = True
+        if groups:
+            group_check = any(x in tool_str for x in groups.split(","))
+        else:
+            group_check = "default" in tool_str
+        if remotes:
+            remote_check = any(x in tool_str for x in remotes.split(","))
+        if tools:
+            tool_check = any(x in tool_str for x in tools.split(","))
+
+        if tool_check and remote_check and group_check:
+            assert not tool.exists()
+            # Default group should exist even if all the remotes are gone
+            if "default" in tool_str:
+                assert tool.parent.parent.exists()
+            # if custom group exists then we need to make sure at least one
+            # of the remote is not empty
+            elif tool.parent.parent.exists():
+                for remote_dir in tool.parent.parent.iterdir():
+                    assert any(
+                        remote_dir.iterdir()
+                    ), f'Found an empty "remote" directory, "{remote_dir!s}", which should have been deleted.'
+        else:
+            assert tool.exists()
+
+
+@pytest.mark.parametrize(
+    "groups", ["default", "default,wrong_group", "group1,wrong_group"],
+)
+@pytest.mark.parametrize(
+    "remotes", ["", "remote1,wrong_remote1", "wrong_remote1,wrong_remote2"],
+)
+@pytest.mark.parametrize(
+    "tools", ["", "pidstat,mpstat,wrong_tool1", "wrong_tool1,wrong_tool2"],
+)
+def test_clear_tools_test86(tools_configuration, groups, remotes, tools):
+    """ Attempt to clear tools at various group-remote-tool locations where the
+    combination may not present on the hosts"""
+    command = [
+        "pbench-clear-tools",
+        f"--groups={groups}" if groups else "",
```

Comment:
You now have a blank value for `remotes` and `tools`, but not for `groups` so this ternary is still unnecessary.

I hesitate to say I prefer the new logic in 85; on one hand it doesn't leave an empty array element like this sequence, on the other it's a bit harder to read. Still, you don't *need* logic for this one, so I'd rather have none. 😆

Reply:
Ah I forgot to update this test file with that, Thanks!